File: Part 2 - Regression/Section 6 - Polynomial Regression/regression_template.R
#Regression Template

#Importing the dataset
dataset= read.csv('Position_Salaries.CSV')
dataset=dataset[2:3]

#splitting the dataset into the Training set and Test set 
#install.packages('caTools')
#library(caTools)
#set.seed(123)
#split = sample.split(dataset$Purchased,SplitRatio = 0.8)
#training_set=subset(dataset,split==TRUE)
#test_set=subset(dataset,split==FALSE)

#Feature Scaling 
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3]=scale(test_set[, 2:3])

#Fitting The Regression Model to tht dataset 
#Create regressor here 

#Predict a new result 
y_pred =  predict(regressor,data.frame(Level = 6.5))

#visualising the Regression  Model result 
library(ggplot2)
ggplot()+
  geom_point(aes(x = dataset$Level ,y =dataset$Salary),
             colour = 'red')+
  geom_line(aes(x = dataset$Level, y = predict(regressor, newdata = dataset)),
            colour = 'blue')+
  ggtitle('Truth or Bluff(Regression Model)')+
  xlab('Level')+
  ylab('Salary')

#visualising the Regression  Model result (for higher resolution and smoother curve)
library(ggplot2)
x_grid = seq(min(dataset$Level),max(dataset$Level),0.1)
ggplot()+
  geom_point(aes(x = dataset$Level ,y =dataset$Salary),
             colour = 'red')+
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue')+
  ggtitle('Truth or Bluff(Regression Model)')+
  xlab('Level')+
  ylab('Salary')

File: Part 2 - Regression/Section 6 - Polynomial Regression/regression_template.py
#Regression_template

"""
Created on Tue Mar 21 19:41:41 2017

@author: ali5c_000
"""
# Importing the libraries 
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd 
#Importing the dataset

dataset= pd.read_csv('Position_Salaries.csv')
X= dataset.iloc[:,1:2].values
y=dataset.iloc[:,2].values

#splitting the dataset into the Training set and Test set 
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, Y_test= train_test_split(X,y, test_size = 0.2, random_state=0)

#Feature Scaling 
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)"""
              
#Fitting  Regression Model to the Dataset
#create your regressor here 

#Predicting result 
y_pred = regressor.predict(6.5)

#Visuling the Regression results 
plt.scatter(X,y, color =  'red')
plt.plot(X,regressor.predict(X), color = 'blue')
plt.title('Truth or Bluff (Regression Model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

#Visualing the Regression result (for higher resolution and smother curve)
X_grid = np.arange(min(X),max(X),0.1)
X_grid = X_grid.reshape(len(X_grid),1)
plt.scatter(X,y, color =  'red')
plt.plot(X_grid,regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Regression Model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()